### PR TITLE
Added cloud_storage search module for issue #109

### PR DESCRIPTION
**Pull request by vikas-kundu**
Also added amazon s3 to the list. Here is the updated gif: https://asciinema.org/a/0hMBRbUQuUqLj8hsWkzw7tYmf

### CODE REVIEW

File: modules/osint/cloud_storage.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+meta = {
+	'name': 'Cloud Storage',
+	'author': 'Vikas Kundu',
+	'version': '0.5',
```
Comment (saeeddhqan) [inline modules/osint/cloud_storage.py]: How?

File: modules/osint/cloud_storage.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	'description': 'Search cloud storage sites and show the results.',
```
Comment (saeeddhqan) [inline modules/osint/cloud_storage.py]: Better than this description? Express the sources that the module explores.

File: modules/osint/cloud_storage.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		'yippy_q': f'"{site_url}" {query}',
+		'millionshort_q': f'site:{site_url} "{query}"',
+		'qwant_q': f'site:{site_url} {query}'
+	}
+	
+	self.thread(search, self.options['thread'], engine, query, q_formats, limit, count, meta['sources'])
+
+	output['site'] = site_name
+	output['links'] = list(self.reglib().filter(r"https?://([\w\-\.]+\.)?"\
+	+site_url.replace('.','\.')+"/", list(set(LINKS)))) #escaping . for regex search using replace()
```
Comment (saeeddhqan) [inline modules/osint/cloud_storage.py]: spaces.

File: modules/osint/cloud_storage.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+	self.save_gather(output, 'osint/cloud_storage', query, output=self.options.get('output'))
+	return output
+
+def module_run(self):
+	sites = {'GoogleDrive': 'drive.google.com', 'DropBox': 'dl.dropboxusercontent.com',\
```
Comment (saeeddhqan) [inline modules/osint/cloud_storage.py]: declare it on the module_api. It causes an error when the user uses the --api option.

File: modules/osint/cloud_storage.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	 'OneDrive': '1drv.ms', 'Box': 'box.com/s', 'Amazon S3': 's3.amazonaws.com'}
+
+	for site_name,site_url in sites.items():
```
Comment (saeeddhqan) [inline modules/osint/cloud_storage.py]: Do it on the module_api. Module run just shows the results.

File: modules/osint/cloud_storage.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+meta = {
+	'name': 'Cloud Storage',
+	'author': 'Vikas Kundu',
+	'version': '0.5',
+	'description': 'Search cloud storage sites and show the results.',
+	'sources': (	'google', 'carrot2', 'bing', 'yippy', 'yahoo', 'millionshort', 'qwant', 'duckduckgo'),
```
Comment (saeeddhqan) [inline modules/osint/cloud_storage.py]: correct the spaces.

File: modules/osint/cloud_storage.py (hunk 1 above)
Comment (vikas-kundu) [reply modules/osint/cloud_storage.py]: What changes need to be made here? The name?

**Comment by saeeddhqan** [comment]
Check the amendments and let me know if it works.

**Comment by vikas-kundu** [comment]
Yes, it works. Made some minor changes like correcting the grammar of description and turning sites into a list instead of dict. Here is a gif of the same: https://asciinema.org/a/XXNRi0G6zhYqqmaybmBDnaA5u

**Pull request merged**